subsys/logging/log_backend_fs.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <logging/log_backend.h>
+#include "log_backend_std.h"
+#include <assert.h>
+#include <fs/fs.h>
+#include <fs/littlefs.h>
+
+#define MAX_PATH_LEN 256
+#define MAX_FLASH_WRITE_SIZE 256
+#define LOG_PREFIX_LEN (sizeof(CONFIG_LOG_BACKEND_FS_FILE_PREFIX) - 1)
+
+enum backend_fs_state {
+	BACKEND_FS_NOT_INITIALIZED = 0,
+	BACKEND_FS_CORRUPTED,
+	BACKEND_FS_OK
+};
+
+FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(storage);
+
+static struct fs_mount_t lfs_storage_mnt = {
+	.type = FS_LITTLEFS,
+	.fs_data = &storage,
+	.storage_dev = (void *)FLASH_AREA_ID(storage),
+	.mnt_point = CONFIG_LOG_BACKEND_FS_MOUNT_POINT,
+};
+
+static char fname[MAX_PATH_LEN];
```

Comment:
You could save some RAM here by only reserving the amount required for the prefix and the sequence number,